cmd/bg-prov/README.md

```diff
@@ -1 +1,334 @@
+            Prints more information about ./bg-prov
+```
+Every subcommand has several required or optional arguments and flags. To learn more about them:
+```bash
+./bg-prov <subcommand> -h
+```
+
+Extended documentation about subcommands:
+--------------
+
+```bash
+./bg-prov show-km       Prints Key Manifest binary in human-readable format
+        <path>  Path to binary file containing Key Manifest
+```
+
+```bash
+./bg-prov show-bpm      Prints Boot Policy Manifest binary in human-readable format
+        <path>  Path to binary file containing Boot Policy Manifest
+```
+    
+```bash
+./bg-prov show-acm      Prints ACM binary in human-readable format
+        <path>  Path to binary file containing Authenticated Code Module (ACM)
+```
+
+```bash
+./bg-prov show-all      Prints BPM, KM, FIT and ACM from Firmware image binary in human-readable format
+        <path>  Path to full Firmaware image binary file containing Key Manifest, Boot Policy Manifest and ACM
+```
+    
+```bash 
+./bg-prov export-acm    Exports ACM binary from Firmware image into file
+        <bios>    Path to the full Firmware image binary file.
+        <out>     Path to the newly generated ACM binary file.
+```
+   
+```bash
+./bg-prov export-km     Exports KM structures from Firmware image image into file
+        <bios>    Path to the full Firmware image binary file.
+        <out>     Path to the newly generated Key Manifest binary file.
+```
+    
+```bash
+./bg-prov export-bpm    Exports BPM structures from Firmware image image into file
+        <bios>    Path to the full Firmware image binary file.
+        <out>     Path to the newly generated Boot Policy Manifest binary file.
+```
+ 
+```bash
+./bg-prov read-config   Reads config from existing BIOS file and translates it to a JSON configuration
+        <config>    Path to the JSON config file.
+        <bios>      Path to the full Firmware image binary file.
+```
+
+        
+```bash
+./bg-prov km-gen        Generate KM file based of json configuration
+        <km>     Path to the newly generated Key Manifest binary file.
+        <key>    Public Boot Policy signing key
+
+        --config=STRING                  Path to the JSON config file.
+        --revision=UINT-8                Platform Manufacturer’s BPM revision number.
+        --svn=UINT-8                     Boot Policy Manifest Security Version Number
+        --id=UINT-8                      The key Manifest Identifier
+        --pkhashalg=UINT-16              Hash algorithm of OEM public key digest
+        --bpmpubkey=STRING               Path to bpm public signing key
+        --bpmhashalgo=ALGORITHM          Hash algorithm for bpm public signing key
+        --out=STRING                     Path to write applied config to
+        --cut                            Cuts the signature before writing to binary (Facebook requirement)
+```
+ 
+```bash
+./bg-prov bpm-gen             Generate BPM file based of json configuration and full bios image

```

Comment:
the help text mixes bios and firmware. What is actually needed? only the bios region or a complete firmware image?

Reply:
@zaolin

Reply:
Will fix it later on. Complete BIOS